core/classes/Adapter.js

```diff
@@ -104,6 +106,19 @@ module.exports = class Adapter {
     this.schemas[identity.toLowerCase()] = schema
   }
 
+  /**
+   * Validate data using ajv
+   * @param {Object} schema Schema to validate
+   * @param {Object} data Data to validate
+   * @returns {{passed: boolean, errors: Array<ajv.ErrorObject>}} The result of validation
+   * @private
+   */
+  _validate(schema, data) {
+    const validator = ajv.compile(schema)
```

Comment:
Schema doesn't change during the lifetime of app instance. And, compiling schema may take sufficient processor time. So, it's wiser to compile them during initialization and put validators inside models